path_creds_create.go

```diff
@@ -62,7 +63,14 @@ func (b *backend) credsReadHandler(ctx context.Context, req *logical.Request, d
 	}
 	userPrefix := role.UserPrefix
 	if role.UserPrefix == defaultUserPrefix {
-		userPrefix = fmt.Sprintf("%s_%s", role.UserPrefix, req.DisplayName)
+		// Hash display name
+		h := sha256.New()
+		_, err := h.Write([]byte(req.DisplayName))
```

Comment:
if the intent is to hide the user name, we might want to salt the hash, otherwise it's easy to check hashes against easily guessable user names.